Formula/aws-auth.rb

```diff
@@ -0,0 +1,26 @@
+require "language/node"
+
+class AwsAuth < Formula
+  desc "Allows you to programmatically authenticate into AWS accounts through IAM roles"
+  homepage "https://github.com/iamarkadyt/aws-auth#readme"
+  url "https://registry.npmjs.org/@iamarkadyt/aws-auth/-/aws-auth-2.1.2.tgz"
+  sha256 "570f767500ed75aa3855338914a4bd2b1ddc030c5b9574fc257b1c59150380c4"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink libexec.glob("bin/*")
+  end
+
+  test do
+    assert_predicate bin/"aws-auth", :exist?
+    assert_predicate bin/"aws-auth", :executable?
+
+    run_output = shell_output("#{bin}/aws-auth 2>&1", 1)
+    assert_match "aws-auth help", run_output
+    assert_match "Use this command to show this help message.", run_output
+    assert_match "Expected to be invoked with a command but got none", run_output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Maintainer here. This library does need credentials to operate. It has a dependency on user-added AWS accounts that it hooks into. Some ideas for tests:
* Test `aws-auth config` command, and that a simple configuration can be added properly.
* Test that `aws-auth login` invokes an interactive prompt asking to decrypt the configuration file, fails with an error when configuration is missing, or fails if the password is incorrect.
* Test `aws-auth reset` command that erases the configuration file.

Reply:
Hi @SMillerDev, as @iamarkadyt all the commands require interactivity (also reset).
I'm actually trying to test `aws-auth login` without luck, at the moment I did not find a way to use `shell_output` or `pipe_output` with that kind of interactive prompt.

Reply:
I've updated the test using pty and simulating an interactive session with `aws-auth login`.
@SMillerDev do you think that it's enought?

Reply:
Yeah, this test makes sense to me, thanks @DocLM !